build.yaml

```diff
@@ -0,0 +1,4 @@
+webhookVersion: 103.0.0+up0.4.0
+cspAdapterMinVersion: 103.0.0+up3.0.0
+defaultShellVersion: rancher/shell:v0.1.22
+fleetVersion: 103.1.0+up0.9.0-rc.5
```

Comment:
Can we have the keys in this YAML file directly correspond to environment variable names? That way we don't need one more map of what value here corresponds to a value in the Dockerfile.

Reply:
Also, why does this have to be a YAML file? Can't we just make this a file like:

```
WEBHOOK_VERSION=103.0.0+up0.4.0
CSP_ADAPTER_VERSION=103.0.0+up3.0.0
...
```

And just directly source this in the script execution?

Reply:
If we plan to stick with YAML, can we also move other things like https://github.com/rancher/rancher/blob/58bfcb19605d51bf24f0aa5bf1d60a68b1ae2a27/pkg/image/origins.go#L21-L252 into this file?

It would be nice if we had a format like:

```yaml
env:
  WEBHOOK_VERSION: 103.0.0+up0.4.0
  ...
image-origins:
  aci-containers-controller: https://github.com/noironetworks/aci-containers
  ...
```

Reply:
1. Agreed on making the keys correspond to environment variable names, will adjust.
2. It doesn't need to be YAML; YAML is just a decent, natural option. And, like your nested example shows, picking YAML is good for the future, should we decide to have more a complicated structure of keys.
3. As for the `origins.go` file, we can do this, but perhaps in a separate PR. This PR aims to introduce the new system and migrate only a handful of variables (listed in PR description).

Reply:
If we plan to introduce a more complicated structure of keys, I think the keys we introduce here should not be top-level keys.